src/sys/wait.rs

```diff
@@ -240,19 +245,27 @@ pub fn wait() -> Result<WaitStatus> {
     waitpid(None, None)
 }
 
-/// Returns an iterator over all currently available events of child processes, as would be available via waitpid with the 
+/// Returns an iterator over all currently available events of child
+/// processes, as would be available via waitpid with the
 /// [`WNOHANG`](struct.WaitPidFlag.html#associatedconstant.WNOHANG) flag.
 ///
-/// This iterator is best used with a signal handler on the signal `SIGCHLD` to handle all pending events. The iterator ends either
-/// if all remaining children are still running normally or no more children are running.
-/// 
-/// Note that you should not call this directly in a signal handler, since those have some restrictions on what may
-/// happen in them. Use a conditional variable to notify some other routine or use a crate like [signal_hook](https://crates.io/crates/signal-hook)
+/// This iterator is best used with a signal handler on the signal `SIGCHLD`
+/// to handle all pending events. The iterator ends either
+/// if all remaining children are still running normally or no more children
+/// are running.
+///
+/// Note that you should not call this directly in a signal handler, since
+/// those have some restrictions on what may
+/// happen in them. Use a conditional variable to notify some other routine
+/// or use a crate like [signal_hook](https://crates.io/crates/signal-hook)
 /// ```
```

Comment:
I think you need to put this in an `# EXAMPLES` section so rustdoc will render it appropriately.  You'll also want to add `no_run` so it won't hang during `cargo test`

Reply:
It should not hang, because it exits immediately if there are no events available but yeah it should not run, just compile. I'll add this.